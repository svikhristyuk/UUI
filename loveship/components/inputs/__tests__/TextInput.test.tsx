import React from "react";
import renderer from "react-test-renderer";
import {TextInput} from "../TextInput";
import { ReactComponent as AcceptIcon } from "../../icons/accept-12.svg";

describe("TextInput", () => {
    const value = "test";
    const onChange = jest.fn();

    it("should be rendered correctly", () => {
        const tree = renderer
            .create(<TextInput value={ value } onValueChange={ onChange }/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should be rendered correctly with extra props", () => {
        const tree = renderer
            .create(<TextInput
                value={ value }
                onValueChange={ onChange }
                onAccept={ jest.fn() }
                onCancel={ jest.fn() }
                icon={ AcceptIcon }
                iconPosition='right'
                isDropdown
                isOpen
                size='60'
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});