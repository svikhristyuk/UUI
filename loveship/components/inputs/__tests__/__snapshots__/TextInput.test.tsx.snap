// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should be rendered correctly 1`] = `
<div
  className="container uui-input-box -clickable colorSky root size-36 mode-form"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input line-height-18 font-size-14 v-padding-8"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="test"
  />
</div>
`;

exports[`TextInput should be rendered correctly with extra props 1`] = `
<div
  className="container uui-input-box -clickable colorSky root size-60 mode-form"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input line-height-30 font-size-24 v-padding-14"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="test"
  />
  <div
    className="container uui-icon uui-enabled -clickable uui-icon-accept"
    onClick={[MockFunction]}
    style={Object {}}
  >
    <SvgrURL
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled -clickable uui-icon-cancel -clickable"
    onClick={[Function]}
    style={Object {}}
  >
    <SvgrURL
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <SvgrURL
      className=""
    />
  </div>
  <div
    className="container uui-icon uui-enabled -clickable"
    style={Object {}}
  >
    <SvgrURL
      className="flipY"
    />
  </div>
</div>
`;
